cranelift/codegen/meta/src/shared/instructions.rs

```diff
@@ -3619,6 +3616,12 @@ pub(crate) fn define(
         ]),
     );
 
+    let NarrowInt = &TypeVar::new(
+        "NarrowInt",
+        "An integer type of width up to `i64`",
+        TypeSetBuilder::new().ints(8..64).build(),
+    );
```

Comment:
The other place that `NarrowInt` is used is for `iconst`. I recently asked Chris about whether we should be making use of `iconst`'s documented ability to construct a vector, and he was surprised that it was documented that way. Perhaps we should just remove `simd_lanes` from the original definition of `NarrowInt` instead of shadowing it here?

Reply:
Happy to make that change as well.

Reply:
Apparently a couple of parser filetests use `iconst` with vector types, but since they're only testing the parser I don't think we should take that as a sign that this is actually used.

In `rewrite.clif` I think you can just change it to an `iconst.i32`. Or possibly delete the test in question entirely, because I don't think it's testing anything related to what the file-header comment claims to be testing.

In `tiny.clif` I guess the test is only supposed to check that the parser can parse lane indexes for `extractlane` and `insertlane`. Making `v0` be an `i32x4` parameter instead of the result of an `iconst` should do that just as well, I think.